snuba_sdk/function.py

```diff
@@ -103,3 +105,34 @@ class Function(CurriedFunction):
     initializers: Optional[Sequence[Union[ScalarLiteralType, Column]]] = field(
         init=False, default=None
     )
+
+
+identifier_re = re.compile("^[a-zA-Z_][a-zA-Z0-9_]*$")
+
+
+class InvalidLambdaError(InvalidExpressionError):
+    pass
+
+
+@dataclass(frozen=True)
+class Identifier(Expression):
+    name: str
+
+    def validate(self) -> None:
+        if not isinstance(self.name, str) or not identifier_re.match(self.name):
+            raise InvalidLambdaError(f"`{self.name}` is not a valid identifier")
+
+
+@dataclass(frozen=True)
+class Lambda(Expression):
+    identifiers: Sequence[str]
```

Comment:
Should this be a sequence of Identifier which you defined above?

Reply:
No, I deliberately left it a string so that the construction of a Lambda looks like `Lambda(["x", "y"], ...)` instead of `Lambda([Identifier("x"), Identifier("y")], ...)`. It's unnecessary to define the type there since they can't be anything but identifiers.